StandardCyborgNetworking/StandardCyborgNetworking/Model/ServerScan.swift

```diff
@@ -7,19 +7,47 @@
 
 import Foundation
 
-public struct ServerScan {
+public struct ServerScan: Codable {
     
-    public enum UploadStatus {
+    public enum UploadStatus: RawRepresentable, Codable {
+        public typealias RawValue = String
+
         case notUploaded
         case uploading(percentComplete: Double)
         case uploaded
+
+        public var rawValue: String {
+            switch self {
+            case .notUploaded: return "notUploaded"
+            case .uploading(let percentComplete): return "uploading_\(percentComplete)"
+            case .uploaded: return "uploaded"
+            }
+        }
+
+        public init?(rawValue: RawValue) {
+            if rawValue == UploadStatus.notUploaded.rawValue {
+                self = .notUploaded
+            } else if rawValue == UploadStatus.uploaded.rawValue {
+                self = .uploaded
+            } else if rawValue.starts(with: "uploading_"), let percentComplete = Double(rawValue.split(separator: "_").last!) {
+                self = .uploading(percentComplete: percentComplete)
+            }
+
+            return nil
+        }
     }
     
     public var localUUID = UUID()
     public var key: String?
     public var createdAt: Date?
     public var uploadedAt: Date?
     public var uploadStatus: UploadStatus?
+
+    enum CodingKeys: String, CodingKey {
```

Comment:
What access level does this default to without an explicit keyword? If this is only used within this class, I'd prefer to explicitly mark it `private`

Reply:
I think technically it defaults to `internal` so it will by default be available to the module, but not outside of it. I can change it to private.